### PR TITLE
Added Custom Desired Capabilities Support

### PR DESCRIPTION
**Pull request by jossjacobo**
Added boolean support for custom capabilities
Changed support for custom caps to just not Maps

Separated PR https://github.com/willowtreeapps/conductor-mobile/pull/34 and changed the target merge branch to `develop`

### CODE REVIEW

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 1]
```diff
@@ -385,4 +372,60 @@ public boolean isAutoGrantPermissions() {
...
+
+    private void putCustomCapabilities(Object keyCustomCapabilities) {
+        Map<String, Object> custom = new HashMap<>();
+        if (keyCustomCapabilities instanceof Map) {
+            for (String key : ((Map<String, Object>) keyCustomCapabilities).keySet()) {
+                Object value = ((Map) keyCustomCapabilities).get(key);
+                if (!(value instanceof Map)) {
+                    custom.put(key, value);
+                } else {
+                    throw new ClassCastException(String.format("%s cannot be of type Map", value));
```
Comment (konwinkler) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Why are explicitly forbidding `Map` type values?

Comment (jossjacobo) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: So what this prevents is the usage of nested maps in the `customCapabilities` field, like this:
```yaml
custom:
  customCapabilities:
    foo: foo
    fizz:
      bar: baz
      fazz: kazz
```
We might want to support this at some point, but currently I didn't see any need for supporting it, makes things a bit simpler.

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 2]
```diff
@@ -10,11 +10,13 @@
 import java.net.URL;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class ConductorConfig {
     private static final String DefaultConfigFile = "/config.yaml";
+    private static final String CUSTOM_CAPABILITIES = "customCapabilities";
```
Comment (fuzzybinary) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: We have to decide if we want SHOUTY_CAPS or CamelCaps for constants....

Comment (jossjacobo) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: [SHOUTY_IS_USUALLY_THE_CONVENTION_FOR_CONSTANTS](https://www.safaribooksonline.com/library/view/java-8-pocket/9781491901083/ch01.html#generic_type_parameter_names)

Comment (fuzzybinary) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Yeah but it's awful.

Comment (jossjacobo) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: What? You need to speak louder, I_CAN'T_HEAR_WHAT_YOU_ARE_SAYING!

**Pull request merged**